lorien/Misc/Settings.gd

```diff
@@ -61,4 +67,43 @@ func set_value(key: String, value = null):
 	_config_file.set_value(DEFAULT_SECTION, key, value)
 	_save_settings()
 	
+# -------------------------------------------------------------------------------------------------
+func _setup_default_shortcuts() -> void:
+	var new_actions := []
+	for action_name in Utils.bindable_actions():
+		if ! _config_file.has_section_key(SHORTCUTS_SECTION, action_name):
+			new_actions.append(action_name)
+	
+	if len(new_actions) > 0:
+		var old_actions := []
+		for action_name in Utils.bindable_actions():
+			if ! action_name in new_actions:
+				old_actions.append(action_name)
+
+		for new_action in new_actions:
+			var event_list := InputMap.get_action_list(new_action)
+			for event in event_list:
+				for old_action in old_actions:
+					if InputMap.action_has_event(old_action, event):
+						event_list.erase(event)
+						break
+			_config_file.set_value(SHORTCUTS_SECTION, new_action, event_list)
```

Comment:
This is the "forwards compatiblity"-bit. 

# Problem

As Lorien is developed further, it is likely that new actions will be introduced with new default keybindings. The keybindings dialog makes sure that duplicate keybindings are not allowed, but if the software adds a new action with a key that the user already has assigned, it will cause a duplicate keybinding.

# Solution

This code sorts actions into "new actions", ones that are not yet mentioned in the settings-file, and "old actions", ones that are mentioned in the settings file. Then it filters all InputEvents assigned to the "new actions" by comparing them to the events assigned to "old actions" and removes all actions that were assigned in the existing settings-file to an "old action".

This way, we prevent duplicate keybindings from occurring when Lorien is updated, and also annoyance of users if their carefully assigned input maps being modified during updates.

Reply:
Yep, good thinking! Thanks for explaining.